### DISCUSSION TITLE
when the player is active, run command

### DISCUSSION THREAD

**Discussion by myavuzokumus** [Zrips/CMI#4327] [Feature Requests]
Can you add a setting that will allow the vote to be counted and commands run when the player is only active on the server?
VotesOnlyPlayerOnline: true/false

**Comment by CloudeLecaw** [comment]
Why it only should be counted when he is online? You can vote on the sites and dont need to be online a vote is a vote if your online or not. So its good that every vote counts. online or not.
If votes only count when your online what you think will happen?
Your players see its not counted because they not online and they go online for voting?.. What about voting while on the way..
No this will just result in that less people will vote for the server anymore.

really no good idea.
Also you cant let run commands only when player is online or only offline of the sever
these comamnd are called....
supprice...
```
ifonline [playerName] (command)
Perform command only when player is online
```

```
ifoffline [playerName] (command)
Perform command only when player is offline
```

So what you can do is create a reward command that only et executed if player is online or offline. But that stores nothing and does not give it a player later when he is online. if he is not online/offline depending on the command then still nothing will happen and the player voted for "nothing" as reward... well this is not the best solution you should choose...

For me it sound like ..what you want is: a command that get executed when player gets online after vote + when inventory is not full..
cmi has a simple vote listener and with this this is not possible what you want.
For your requirment u may need something like the voteroulette.

**Comment by myavuzokumus** [comment]
Actually, this is not the reason why I wanted this, the safe plugin I used can only give the key virtually if the player is online, if not, it cannot. And when an absent player votes on the server, I don't want it to appear that they are voting in the chat. Because we have 5 voting sites and giving them all at once causes pollution of conversation.

**Comment by myavuzokumus** [comment]
Actually, as you said, I can fix this request with these commands, but this time it might be better if it can bring a command about "when the player is active". This way, when the player enters the server, he can get the vote case.

**Comment by CloudeLecaw** [comment]
This cant be done by the CMI used vote listener that is a simple GAL listener. For this you need a plugin that stores the vote and execute command when player gets online. some Votelistener plugins like [voteroulette](https://www.spigotmc.org/resources/voteroulette.8126/) support this.
```
Offline voting support! If a player votes for the server but isn't in game to receive rewards, the rewards will be there waiting for them to claim the next time they log on.
Option to have awards auto-claimed when a player logs in!
```

**Comment by myavuzokumus** [comment]
I don't want it as a voting feature, I want it as a command.
/whenonline [player] [command]
Also my server version its 1.16.3.

**Comment by CloudeLecaw** [comment]
hm this would still need a complete new database entry where these "is not online" or storage rewards will stored if player is not online...

to be honest i dont think this will be added so soon... but yes you can suggest it.. why not. the idea itself is nice.

but for only execute when player is onnline you could use `ifonline [playerName] (command)` this just means nothing happens when he is not online. and it also will not happen later when the player gets on...

until it is added my one time ...i really recomment using a 3rd plugin for that like the voteroulette

**Comment by Zrips** [comment]
@OverBrave 
Do you want to perform command only if player is online or do you want to post pone commands until player logs in? Because those are completely different things and first one is already possible while second one might require extra entry in database with completely different handling of it

**Comment by myavuzokumus** [comment]
I'm talking about the 2nd one. 
Commands will not be executed while the player is not on the server, but will be executed when he enters the server.

**Comment by CloudeLecaw** [comment]
A lot of command work fine if the player is offline, like giving diamonds, or give money or give item value from other plugins or increase tfly time with cmi. Most cmi things work while player is offline too.
Problem is only with plugins that just work with their items and value if player is online. Looks like OverBrave use some of them. And therefor the plugins dev would need to make his plugin compatible with offline player support or cmi would need this extra database

On our server for example, we give players money boxes, diamonds and mystery dust. These can be given to offline players without any problem. The moneybox was not able to give to offline players so we designed a kit that contains the moneybox and this kit we give with cmi to the player. So we used the kit as "bridge" for offline Support of normaly not supported offline give aways.

**Comment by myavuzokumus** [comment]
Yes, I was using a kit for this before, but I cannot succeed because the plugin key I use transmits virtually. And this plugin maker said he didn't have time to do this.

**Comment by CloudeLecaw** [comment]
so lets hope that Zrips has time for this ;-)
